docs/initializing.rst

```diff
@@ -555,6 +556,42 @@ point into the system. It will handle the setup and initialization of the config
 and start up a Toolkit session once all the required pieces have been initialized and set up.
 
 
+Distributing application code from your site
+============================================
+
+If you want to distribute custom application code that may only be accessible for
+users on your local network or from developers who have access keys to the Git
+repositories, you can upload individual bundles to Shotgun and then take over the ``populate_bundle_cache_entry`` and ``can_cache_bundle`` methods of the
+``core/bootstrap.py`` hook.
+
+Here's a suggested setup:
+
+.. image:: ./resources/initializing/bundle_custom_entity.png
+
+As you can see, a non project custom entity has been configured to store Toolkit bundles. The
+most important column here is Descriptor, which is the field the bootstrap hook will use
+to determine if a bundle is available in Shotgun or not.
+
+Once the bundles have been uploaded, you can implement the ``core/bootstrap.py`` hook.
+
+.. literalinclude:: ../tests/fixtures/descriptor_tests/with_create_descriptor_hook/core/hooks/bootstrap.py
```

Comment:
This is really cool. It takes a file and includes everything between the two `#` markers.